cmd/extensions/main.go

```diff
@@ -170,6 +171,21 @@ func main() {
 		health = healthcheck.NewHandler()
 	}
 
+	podReady = true
+	health.AddReadinessCheck("agones-extensions", func() error {
+		if !podReady {
+			return errors.New("Pod is not in Ready state")
+		}
+		return nil
+	})
+
+	signals.NewSigTermHandler(func() {
+		podReady = false
+		time.Sleep(30 * time.Second)
```

Comment:
make this a configurable flag. Personally I would consider defaulting it from helm - I say that because I think it's not hard to make it something like `periodSeconds * failureThreshold * 2`, where the *2 is just a safety/fudge factor.